src/signature-help.jl

```diff
@@ -0,0 +1,354 @@
+using .JS
+using .JL
+
+# initialization
+# ==============
+
+signature_help_options() = SignatureHelpOptions(;
+    triggerCharacters = ["(", ",", ";"],
```

Comment:
```suggestion
    triggerCharacters = ["(", ",", ";", "="],
```
Maybe we should add `=` too as it allows us to get sighelp on cases like
<img width="752" alt="image" src="https://github.com/user-attachments/assets/9dda5dab-a624-462c-97cf-9aa7321213fc" />
?

Reply:
By the way, as a hack that is only possible with VSCode, when experimenting with various `triggerCharacters`, the hack explained at
https://github.com/aviatesk/JETLS.jl?tab=readme-ov-file#dynamic-registration
might be useful.

Reply:
> I think it would make sense to put `=` (and maybe more) in `retriggerCharacters` instead.

The reason I think `triggerCharacters` might be better than `retriggerCharacters` is that currently, as per https://github.com/aviatesk/JETLS.jl/pull/34#issuecomment-2935837838, when typing up to `func(arg1, arg2; k|)`, the signature help gets canceled, so unless we trigger fresh signature help, it might not be retriggered.
At least in VSCode, once signature help disappears, it doesn't seem to be retriggered. However, eglot might handle this case better.

So, if https://github.com/aviatesk/JETLS.jl/pull/34#issuecomment-2935837838 can be resolved, it might be fine to include `=` in `retriggerCharacter`. But registering it into `triggerCharacters` seems to be more robust and non-efficient solution.

> What does Zed do?

The situation with Zed is even worse...
The implementation of signature help in Zed is very primitive, and it doesn't seem to respect `triggerCharacters` at all. I've submitted this as an issue.
https://github.com/zed-industries/zed/issues/31846

Reply:
> if https://github.com/aviatesk/JETLS.jl/pull/34#issuecomment-2935837838 can be resolved

Yup, I'm going to try and exclude `k` from the filtering logic here, as it doesn't make sense to filter on partially-typed kwargs.  I'll test if `=` works well in retriggerCharacters then.

> The situation with Zed is even worse

nooooo.  Thanks for filing the issue haha